vs/client/src/extension.ts

```diff
@@ -1,220 +1,149 @@
-
-import * as path from 'path';
-import { workspace, ExtensionContext, window, commands, Uri, WorkspaceFolder } from 'vscode';
-
-import {
-	LanguageClient,
-	LanguageClientOptions,
-	ServerOptions,
-	TransportKind
-} from 'vscode-languageclient/node';
-import { fixProject, generateBuildFile, reloadProject, setClient } from './requests';
-import { getProjectExplorer, getProjectManager, loadIBMiProjectExplorer } from './ProjectExplorer';
-import { ILEObjectTreeItem, ObjectsView } from './views/objectView';
 import { IProject } from '@ibm/vscode-ibmi-projectexplorer-types/iproject';
-import { ImpactView } from './views/impactView';
-import { getDeployGitFiles as getChanged, getDeployGitFiles as getChangedFiles, getGitAPI, lastBranch } from './git';
-import { initialiseTaskProvider } from './tasks';
+import { commands, ExtensionContext, window, workspace, WorkspaceFolder } from 'vscode';
+import { SourceOrbitApi } from './api';
 import { EnvironmentManager } from './environmentManager';
+import { GitEventHandler } from './gitEventHandler';
+import { LanguageClientManager } from './languageClientManager';
+import { getProjectManager, loadIBMiProjectExplorer } from './ProjectExplorer';
+import { SourceOrbitTask } from './tasks';
+import { ImpactView } from './views/impactView/impactView';
+import { ILEObjectTreeItem } from './views/projectExplorer/ileObjectTreeItem';
+import { SourceOrbitTreeItem } from './views/projectExplorer/sourceOrbitTreeItem';
 
-let client: LanguageClient;
+export async function activate(context: ExtensionContext) {
+	console.log('Congratulations, your extension "vscode-sourceorbit" is now active!');

```

Comment:
Is this good practice?  I am asking not telling.

Reply:
Are you referring to the `let client: LanguageClient;`? I removed this in favor of having a singleton design pattern with `LanguageClientManager`.

If you are referring to the console log, this is just boilerplate code when we created the extension that I added back just to log when the extension is activated. Just used mainly for development in case there are issues with activation.